server/sonar-process-monitor/src/test/java/org/sonar/application/cluster/AppStateClusterImplTest.java

```diff
@@ -116,6 +121,41 @@ public void simulate_network_cluster() throws InterruptedException {
     }
   }
 
+  @Test
+  public void appstateclusterimpl_must_set_sonarqube_version() {
+    TestAppSettings settings = newClusterSettings();
+
+    try (AppStateClusterImpl appStateCluster = new AppStateClusterImpl(settings)) {
+      appStateCluster.registerSonarQubeVersion(getSonarqubeVersion());
+
+      HazelcastInstance hzInstance = createHazelcastClient(appStateCluster);
+      assertThat(hzInstance.getAtomicReference(SONARQUBE_VERSION).get())
+        .isNotNull()
+        .isInstanceOf(String.class)
+        .isEqualTo(getSonarqubeVersion());
+    }
+  }
+
+  @Test
+  public void incorrect_sonarqube_version_must_trigger_an_exception() throws IOException, InterruptedException, IllegalAccessException, NoSuchFieldException {
```

Comment:
we usually name such test `registerSonarQubeVersion_throws_ISE_if_initial_version_is_different()`. Note that `throws Exception` is simpler.